src/open_anything_diffusion/simulations/suction.py

```diff
@@ -0,0 +1,843 @@
+    return int1[1] > int2[0] and int2[1] > int1[0]
+
+
+def get_obj_bbox_xy(object_id, sim_id):
+    bboxes = [p.getAABB(object_id, physicsClientId=sim_id)]  # Add the base one.
+    for i in range(p.getNumJoints(object_id, physicsClientId=sim_id)):
+        aabb = p.getAABB(object_id, i, physicsClientId=sim_id)  # Add the links.
+        bboxes.append(aabb)
+    # xmin = np.array(bboxes)[:, :, 0].min()
+    xmin = functools.reduce(lambda a, b: min(a, b[0][0]), bboxes, 0.0)
+    # xmax = np.array(bboxes)[:, :, 0].max()
+    xmax = functools.reduce(lambda a, b: max(a, b[1][0]), bboxes, 0.0)
+    # ymin = np.array(bboxes)[:, :, 1].min()
+    ymin = functools.reduce(lambda a, b: min(a, b[0][1]), bboxes, 0.0)
+    # ymax = np.array(bboxes)[:, :, 1].max()
+    ymax = functools.reduce(lambda a, b: max(a, b[1][1]), bboxes, 0.0)
+    return [[xmin, xmax], [ymin, ymax]]
+
+
+def get_pointcloud(depth, intrinsics):
+    """Get 3D pointcloud from perspective depth image.
+    Args:
+      depth: HxW float array of perspective depth in meters.
+      intrinsics: 3x3 float array of camera intrinsics matrix.
+    Returns:
+      points: HxWx3 float array of 3D points in camera coordinates.
+    """
+    height, width = depth.shape
+    xlin = np.linspace(0, width - 1, width)
+    ylin = np.linspace(0, height - 1, height)
+    px, py = np.meshgrid(xlin, ylin)
+    px = (px - intrinsics[0, 2]) * (depth / intrinsics[0, 0])
+    py = (py - intrinsics[1, 2]) * (depth / intrinsics[1, 1])
+    points = np.float32([px, py, depth]).transpose(1, 2, 0)
+    return points
+
+
+def fetch_mobility_object(category_name, instance_id):
+    mobility_path = "/home/harry/umpnet/mobility_dataset"
+    urdf_path = os.path.join(mobility_path, category_name, instance_id, "mobility.urdf")
+    with open(
+        os.path.join(
+            mobility_path, category_name, instance_id, "object_meta_info.json"
+        ),
+        "r",
+    ) as f:
+        meta_info = json.load(f)
+    orientation = meta_info["orientation"]
+    offset = [0, 0, meta_info["offset_z"]]
+    scale = meta_info["scale"]
+    return (
+        orientation,
+        offset,
+        scale,
+        meta_info["moveable_link"],
+    )
+
+
+def is_overlapping(int1, int2):
+    return int1[1] > int2[0] and int2[1] > int1[0]
+
+
+def check_collision_placing(ranges, bbox, pos_x, pos_y):
+    if not ranges:
+        return False
+    else:
+        xbound = [
+            pos_x - (bbox[0][1] - bbox[0][0]) / 2,
+            pos_x + (bbox[0][1] - bbox[0][0]) / 2,
+        ]
+        ybound = [
+            pos_y - (bbox[1][1] - bbox[1][0]) / 2,
+            pos_y + (bbox[1][1] - bbox[1][0]) / 2,
+        ]
+        for obj in ranges:
+            obj_xbound = ranges[obj][0]
+            obj_ybound = ranges[obj][1]
+            if is_overlapping(xbound, obj_xbound) and is_overlapping(
+                ybound, obj_ybound
+            ):
+                return True
+
+        return False
+
+
+def compute_flow(
```

Comment:
you should import from flowbot https://github.com/r-pad/flowbot3d/blob/main/flowbot3d/datasets/flow_dataset.py#L17C29-L17C29